src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/test_rdbms_flexible_commands.py

```diff
@@ -1146,6 +1146,74 @@ def _test_flexible_server_replica_mgmt(self, database_engine, resource_group, vn
+
+    def _test_flexible_server_virtual_endpoint_mgmt(self, database_engine, resource_group):
+        location = self.postgres_location
+        primary_role = 'Primary'
+        replica_role = 'AsyncReplica'
+        master_server = self.create_random_name(SERVER_NAME_PREFIX, 32)
+        replica = self.create_random_name(F'azuredbclirep', SERVER_NAME_MAX_LENGTH)
+        master_vnet_args = '--public-access none'
+        virtual_endpoint_name = self.create_random_name(F'virtual-endpoint', 32)
+        read_write_endpoint_type = 'ReadWrite'
+
+        # create a server
+        self.cmd('{} flexible-server create -g {} --name {} -l {} --storage-size {} {} --tier GeneralPurpose --sku-name Standard_D2s_v3 --yes'
+                 .format(database_engine, resource_group, master_server, location, 256, master_vnet_args))
+        primary_result = self.cmd('{} flexible-server show -g {} --name {} '
+                          .format(database_engine, resource_group, master_server),
+                          checks=[JMESPathCheck('replicationRole', primary_role)]).get_output_in_json()
+
+        # create replica
+        self.cmd('{} flexible-server replica create -g {} --replica-name {} --source-server {}'
+                 .format(database_engine, resource_group, replica, master_server),
+                 checks=[
+                     JMESPathCheck('name', replica),
+                     JMESPathCheck('replicationRole', replica_role),
+                     JMESPathCheck('sourceServerResourceId', primary_result['id'])])
+
+        # test virtual-endpoint create
+        self.cmd('{} flexible-server virtual-endpoint create -g {} --server-name {} --name {} --endpoint-type {} --members {}'
+                 .format(database_engine, resource_group, master_server, virtual_endpoint_name, read_write_endpoint_type, replica),
+                 checks=[
+                     JMESPathCheck('endpointType', read_write_endpoint_type),
+                     JMESPathCheck('name', virtual_endpoint_name),
+                     JMESPathCheck('length(virtualEndpoints)', 2)])
+
+        # test virtual-endpoint update
+        update_check = self.cmd('{} flexible-server virtual-endpoint update -g {} --server-name {} --name {} --endpoint-type {} --members {}'
+                 .format(database_engine, resource_group, master_server, virtual_endpoint_name, read_write_endpoint_type, master_server),
+                 checks=[JMESPathCheck('length(members)', 1)]).get_output_in_json()
+
+        # test virtual-endpoint show
+        self.cmd('{} flexible-server virtual-endpoint show -g {} --server-name {} --name {}'
+                 .format(database_engine, resource_group, master_server, virtual_endpoint_name),
+                 checks=[JMESPathCheck('members', update_check['members'])])
+
+        # test virtual-endpoint delete
+        self.cmd('{} flexible-server virtual-endpoint delete -g {} --server-name {} --name {}'
```

Comment:
in this case of direct delete of virtual endpoint it should throw a prompt which is currently not implemented in this PR. For test I think it is better to pass parameter yes here